internal/manifestwork/manifestwork.go

```diff
@@ -175,3 +194,92 @@ func (mwg *manifestWorkGenerator) getOwnedManifestWorks(ctx context.Context, mcm
 
 	return manifestWorkList, err
 }
+
+func (mwg *manifestWorkGenerator) managedClusterKernelMappings(
+	ctx context.Context,
+	mcm hubv1beta1.ManagedClusterModule,
+	kernelVersions []string) []kmmv1beta1.KernelMapping {
+
+	mod := &kmmv1beta1.Module{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      mcm.Name,
+			Namespace: mcm.Spec.SpokeNamespace,
+		},
+		Spec: mcm.Spec.ModuleSpec,
+	}
+
+	logger := log.FromContext(ctx)
+
+	mappings := make([]kmmv1beta1.KernelMapping, 0, len(kernelVersions))
```

Comment:
We are doing this to avoid a situation where several mappings would be added to the `ManifestWork` with the same `regex` field.

> What happens if there is an upgrade on the spoke but the mapping wasn't added yet to the module by the hub?

We can make it clear in the docs that it should be added to the `ManagedClusterModule` first.

Reply:
If there is a spoke OCP upgrade (i.e. a new kernel is available in one or more of its nodes), the following happens:

* the spoke KMM `ClusterClaim` on the `ManagedCluster.Status` is updated by KMM Managed (on the spoke) 
* [as a result](https://github.com/kubernetes-sigs/kernel-module-management/blob/main/controllers/hub/managedclustermodule_reconciler.go#L164-L170) KMM Hub will reconcile the `ManagedClusterModule` targeting that spoke
* this reconciliation will add the new literal `KernelMapping`(s) to the `Module` that will be deployed to the spoke 

Does this make sense?

Reply:
> Does this make sense?

It does. Thanks.